workflows/core_link/create_core_link.py

```diff
@@ -35,17 +37,17 @@
 
 def initial_input_form_generator(product: UUIDstr, product_name: str) -> FormGenerator:
     class SelectNodes(FormPage):
-        class Config:
-            title = f"{product_name} - node A and B"
+        model_config = ConfigDict(title=f"{product_name} - node A and B")
 
         node_subscription_id_a: node_selector("NodesEnumA")  # type:ignore # noqa: F821
         node_subscription_id_b: node_selector("NodesEnumB")  # type:ignore # noqa: F821
 
-        @validator("node_subscription_id_b", allow_reuse=True)
-        def separate_nodes(cls, v: str, values: dict, **kwargs):
-            if v == values["node_subscription_id_a"]:
-                raise AssertionError("node B cannot be the same as node A")
-            return v
+        @field_validator("node_subscription_id_b")
```

Comment:
I think for now this is ok. An improvement might be to have an annotated type NodeSubscriptionA and NodeSubscriptionB, where you could add this validator to NodeSubscriptionB. Alternatively If you want to decouple the two types, an model_validator might make sense.

Reply:
A model_validator seems like the best solution